diff --git a/include/rapidjson/document.h b/include/rapidjson/document.h
index 5ce0919..0d7c9d8 100644
--- a/include/rapidjson/document.h
+++ b/include/rapidjson/document.h
@@ -319,2 +319,0 @@ struct GenericStringRef {
-    GenericStringRef& operator=(const GenericStringRef& rhs) { s = rhs.s; length = rhs.length; }
-
@@ -329,3 +327,5 @@ private:
     GenericStringRef(CharType (&str)[N]) /* = delete */;
+    //! Copy assignment operator not permitted - immutable type
+    GenericStringRef& operator=(const GenericStringRef& rhs) /* = delete */;
 };
